### DISCUSSION TITLE
How to populate multiple select dynamically?

### DISCUSSION THREAD

**Discussion by hrisitjansiwb** [filamentphp/filament#6716] [Help] [answered]
I have following dropdowns

```
 Select::make('country')
                    ->options(Country::where('some' => 'condition')->pluck('name', 'id')
                    ->reactive()

```

And 

```
   Select::make('cities')
                    ->multiple()
                    ->searchable()
                    ->options([])
                    ->reactive()
```

What I am trying to do is when i select country I want to populate cities dropdown. 
One of the solution that is working for me is

```
           Select::make('cities')
                    ->multiple()
                    ->searchable()
                    ->options(function(callable $get) {
                        return City::where(['country_id' => $get('country_id')])->pluck('name', 'id');
                    })
                    ->reactive()
```

The problem with this approach is that I have chart in the form and when I click on the second dropdown the chart disappears (since the html rerender).

Another approach that I tried (but I cannot make it work) is following:

```
 Select::make('country')
                    ->options(Country::where('some' => 'condition')->pluck('name', 'id')
                    ->reactive()
                        ->afterStateUpdated(function($component) {
                        $component->getContainer()->getComponents()[1]->options([
                            '1' => 'demo', '2' => 'another item'
                        ]);
                    })

```

Using this approach I am able to populate some value, but since the field is multiselect the options are not showing (if the field is not multiselect it works okay).

I am pretty much sure that instead calling the $component I can use $set callable too. 

My question is: how to populate **multiselect** field from another field using afterStateUpdated

**Comment by danharrin** [comment, marked as answer]
The chart just needs a `wire:ignore`. Passing a function to options() is correct.

**Reply by hrisitjansiwb** [reply]
That solves the issue actually. Thanks!
